crates/flux-middle/src/rty/expr.rs

```diff
@@ -478,6 +478,7 @@ impl Expr {
             }
             BaseTy::Uint(_) => ExprKind::Constant(Constant::from(bits)).intern(),
             BaseTy::Bool => ExprKind::Constant(Constant::Bool(bits != 0)).intern(),
+            BaseTy::Char => ExprKind::Constant(Constant::from(bits)).intern(),
```

Comment:
This becomes (or something like this, haven't tested it)

```suggestion
            BaseTy::Char => {
                let c = char::from_u32(bits.try_into::<u32>().unwrap()).unwrap();
                ExprKind::Constant(Constant::Char(c)).intern(),
            }
```